### ISSUE TITLE
Can you do ORs or ANDs in a search or filter

### ISSUE THREAD

**Issue by ekristen** [vweevers/level-scout#6]
I don't see any documentation for it, is it possible at this time?

**Comment by vweevers** [comment]
I'm afraid not. My current goal for this module is not to offer a fully fledged query engine, things need to settle first. Right now I'm waiting on `bytespace` to resolve some issues with parent db encoding (bytespace will then replace `level-sublevel` in scout), on `memdown` to publish a new version (with the fix Level/memdown#36) and lastly on mafintosh/sorted-intersect-stream#2 to be merged. After that, I'm want to tackle selectivity factors, find a good abstraction for the query plan, and decouple/modularize the different components of scout. In other words, I won't be adding features (to the public api) for a while.

If you haven't already, you can check out some other [modules](https://github.com/Level/levelup/wiki/Modules) that do what you want.

**Comment by ekristen** [comment]
Understood, your library gets me 90% of what I need.

**Comment by deanlandolt** [comment]
FWIW I suspect we could use the same kind of stream intersection approach in the indexing and build a reasonably simple (if not perfectly efficient) pipeline of stream transforms from the ORs and ANDs.

@vweevers kick the tires on the latest version of `bytespace` I just pushed up -- your test passes (along with some others I created along the same lines), and I went ahead and killed `level-updown` for good, so debugging should be a lot nicer. A `bytespace` is now just a set of `levelup` method wrappers, nothing more. You can create sublevels (and even live streams within these sublevels) in remote multilevel dbs at runtime, without any prior server knowledge (@ekristen IIRC you've been looking for this feature too). I'm still scratching my head on the best way to do remote hooks -- but that's probably out of scope for `level-scout` anyway.

**Comment by vweevers** [comment]
@deanlandolt On remote hooks (in the context of level-scout): I want to leave it to the consumer to set that up. They might only want server-side indexing, or client-side, or full remote madness, or maybe let the host handle indexing and the hooks, but make indexes and search read-accessible over multilevel. It'd be nice to have some tools to make this all easier, but minimal, non-invasive.

Kicked the tires, car broke, so I'm going over to the bytespace repos.

**Comment by vweevers** [comment]
@ekristen Actually, because the optimizer needs work anyway, and it'll be a big rewrite (I want to separate it into a planner that builds trees and an executor that reduces a tree to a single stream), I also started thinking about ORs and explicit ANDs. See #9 if you're interested.
